dvc/output.py

```diff
@@ -563,26 +569,6 @@ def index_key(self) -> Tuple[str, "DataIndexKey"]:
             key = self.fs.path.parts(no_drive)[1:]
         return workspace, key
 
-    def get_entry(self) -> "DataIndexEntry":
-        if self.files and not self.obj:
-            self.obj = self.get_obj()
-
-        entry = DataIndexEntry(
-            meta=self.meta,
-            obj=self.obj,
-            hash_info=self.hash_info,
-        )
-        if self.stage.is_import and not self.stage.is_repo_import:
-            dep = self.stage.deps[0]
-            entry.meta = dep.meta
-            if not entry.obj:
-                if not dep.obj and dep.files:
-                    dep.obj = dep.get_obj()
-                entry.obj = dep.obj
-                if not entry.hash_info and dep.obj:
-                    entry.hash_info = dep.obj.hash_info
-        return entry
```

Comment:
finally this spaghetti is gone. Output is full of other nasty stuff, but looks like data index is settling structure/storage of files.
![image](https://user-images.githubusercontent.com/5367102/217710524-16375f01-81d7-438b-9e77-072b6c957d77.png)